### PR TITLE
network: bridge add support to configure proxy ARP/WIFI

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1876,6 +1876,21 @@
...
+          <listitem>
+            <para>Takes a boolean. Configures whether proxy ARP to be enabled on this port.
+            When unset, the kernel's default will be used.</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term><varname>ProxyARPWiFi=</varname></term>
+          <listitem>
+            <para>Takes a boolean. Configures whether proxy ARP to be enabled on this port
+            which meets extended requirements by IEEE 802.11 and Hotspot 2.0 specifications
```
Comment (yuwata) [inline man/systemd.network.xml]: `specifications.` (missing period.)

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1876,6 +1876,32 @@
...
+            When unset, the kernel's default will be used.</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term><varname>MulticastRouter=</varname></term>
+          <listitem>
+            <para>configure this port for having multicast routers attached. A port with a  multicast
```
Comment (yuwata) [inline man/systemd.network.xml]: `Configures`
and
`with a multicast` (white space is doubled.)

File: man/systemd.network.xml [hunk 3]
```diff
@@ -1876,6 +1876,32 @@
...
+            router will receive all multicast traffic. Takes on of <literal>disabled</literal>
```
Comment (yuwata) [inline man/systemd.network.xml]: `Takes one of`

I prefer `no` instead of `disabled`.

File: man/systemd.network.xml [hunk 4]
```diff
@@ -1876,6 +1876,32 @@
...
+            to disable multicast routers on this port, <literal>query</literal> to let the system detect
+            the presence of of routers, <literal>permanent</literal>to permanently enable multicast traffic
```
Comment (yuwata) [inline man/systemd.network.xml]: `of of`
and
`<literal>permanent</literal> to` (missing white space)

File: man/systemd.network.xml [hunk 5]
```diff
@@ -1876,6 +1876,32 @@
...
+            forwarding on this port or <literal>temporary</literal> to enable multicast routers temporarily
```
Comment (yuwata) [inline man/systemd.network.xml]: `on this port, or` (missing comma)

File: src/network/networkd-link.c [hunk 6]
```diff
@@ -1643,6 +1654,12 @@ static int link_set_bridge(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_BRPORT_PRIORITY attribute: %m");
         }
 
+        if (link->network->multicast_router != _MULTICAST_ROUTER_INVALID) {
+                r = sd_netlink_message_append_u16(req, IFLA_BRPORT_MULTICAST_ROUTER, link->network->multicast_router);
```
Comment (yuwata) [inline src/network/networkd-link.c]: this should be `_append_u8()`. See https://github.com/torvalds/linux/blob/fe5cdef29e41c8bda8cd1a11545e7c6bfe25570e/net/bridge/br_netlink.c#L673

File: src/network/networkd-network.h [hunk 7]
```diff
@@ -84,6 +84,15 @@ typedef enum RADVPrefixDelegation {
         _RADV_PREFIX_DELEGATION_INVALID = -1,
 } RADVPrefixDelegation;
 
+typedef enum MultiCastRouter {
+        MULTICAST_ROUTER_NONE,
+        MULTICAST_ROUTER_TEMPORARY_QUERY,
+        MULTICAST_ROUTER_PERMANENT,
+        MULTICAST_ROUTER_TEMPORARY,
```
Comment (yuwata) [inline src/network/networkd-network.h]: The values are defined in `if_bridge.h`. So, how about the following.
```
typedef enum MulticastRouter {
  MULTICAST_ROUTER_NONE = MDB_RTR_TYPE_DISABLED,
  MULTICAST_ROUTER_TEMPORARY_QUERY = MDB_RTR_TYPE_TEMP_QUERY,
  MULTICAST_ROUTER_PERMANENT = MDB_RTR_TYPE_PERM,
  MULTICAST_ROUTER_TEMPORARY = MDB_RTR_TYPE_TEMP,
} MulticastRouter;
```

File: src/network/networkd-network.h [hunk 8]
```diff
@@ -84,6 +84,15 @@ typedef enum RADVPrefixDelegation {
...
+        _MULTICAST_ROUTER_MAX,
+        _MULTICAST_ROUTER_INVALID = -1,
+} MultiCastRouter;
```
Comment (yuwata) [inline src/network/networkd-network.h]: `Multicast`, instead of `MultiCast`.

File: src/network/networkd-network.h (hunk 7 above)
Comment (ssahani) [reply src/network/networkd-network.h]: yes but that throws errors 
```In file included from ../src/basic/linux/if_bridge.h:19,
                 from ../src/network/networkd-network.h:24,
                 from ../src/network/networkd-manager.h:19,
                 from ../src/network/networkd-dhcp6.c:17:
../src/basic/linux/in6.h:33:8: error: redefinition of ‘struct in6_addr’
   33 | struct in6_addr {
      |        ^~~~~~~~
In file included from ../src/systemd/sd-radv.h:24,
                 from ../src/network/networkd-dhcp6.c:8:
/usr/include/netinet/in.h:211:8: note: originally defined here
  211 | struct in6_addr
      |        ^~~~~~~~
In file included from ../src/basic/linux/if_bridge.h:19,
                 from ../src/network/networkd-network.h:24,
                 from ../src/network/networkd-manager.h:19,
                 from ../src/network/networkd-dhcp6.c:17:
../src/basic/linux/in6.h:41: warning: "s6_addr" redefined
   41 | #define s6_addr   in6_u.u6_addr8
      | 
In file included from ../src/systemd/sd-radv.h:24,
                 from ../src/network/networkd-dhcp6.c:8:
/usr/include/netinet/in.h:219: note: this is the location of the previous definition
  219 | #define s6_addr   __in6_u.__u6_addr8
      | 
In file included from ../src/basic/linux/if_bridge.h:19,
                 from ../src/network/networkd-network.h:24,
                 from ../src/network/networkd-manager.h:19,
                 from ../src/network/networkd-dhcp6.c:17:
../src/basic/linux/in6.h:43: warning: "s6_addr16" redefined
   43 | #define s6_addr16  in6_u.u6_addr16


```

**Comment by ssahani** [comment]
updated thanks for review @yuwata

**Comment by yuwata** [comment]
@ssahani Kernel and glibc's header conflict is now resolved, and the enum values are explicitly specified.

**Comment by ssahani** [comment]
That's Great . Thanks .

**Comment by yuwata** [comment]
(Restarting CIs)

**Comment by ssahani** [comment]
Build is failing now https://travis-ci.org/systemd/systemd/jobs/529523130
```
Werror=shadow -include config.h -pthread -MD -MQ 'systemd-networkd@exe/src_network_networkd.c.o' -MF 'systemd-networkd@exe/src_network_networkd.c.o.d' -o 'systemd-networkd@exe/src_network_networkd.c.o' -c ../src/network/networkd.c
In file included from ../src/basic/linux/if_bridge.h:19:0,
                 from ../src/network/networkd-network.h:5,
                 from ../src/network/networkd-manager.h:19,
                 from ../src/network/networkd.c:14:
../src/basic/linux/in6.h:33:8: error: redefinition of ‘struct in6_addr’
 struct in6_addr {
        ^
In file included from /usr/include/arpa/inet.h:22:0,
                 from ../src/network/networkd-manager.h:4,
                 from ../src/network/networkd.c:14:
/usr/include/netinet/in.h:211:8: note: originally defined here
 struct in6_addr
        ^
In file included from ../src/basic/linux/if_bridge.h:19:0,
                 from ../src/network/networkd-network.h:5,
                 from ../src/network/networkd-manager.h:19,
                 from ../src/network/networkd.c:14:
../src/basic/linux/in6.h:41:0: error: "s6_addr" redefined [-Werror]
 #define s6_addr   in6_u.u6_addr8
 ^

```

**Comment by yuwata** [comment]
Ouch, I did not notice that. Will update.

**Comment by yuwata** [comment]
@keszybz Could you review this? As this PR contains @ssahani's new feature and my header cleanups.

**Comment by ssahani** [comment]
As always looks good .

**Comment by yuwata** [comment]
@ssahani Thanks.

**Pull request merged**